Pyload fails to install on specific Ubuntu 20.04 image.

## Bug description
Pyload fails to install on Ubuntu 20.04, suspected due to specific build.

## Reproduction steps
Steps to reproduce the behavior:
1. Launch new Ampere ARM instance on Oracle Cloud
2. `sudo apt update && sudo apt upgrade`
3. Download and run the install script from README
4. `box install pyload`

## Installed applications through swizzin
pyload

## Expected behavior
Installation completes successfully

## Server Info
**(please complete the following information):**
 - OS: Ubuntu 20.04
 - Arch: arm64
 - Image Source: Oracle Cloud, build 2021.05.13-0

## Is Swizzin up to date?
Uses latest install script.

## Additional context
The install step "Initializing database" fails. The logs reveal:

 ```
 running build_ext
    building '_cffi_backend' extension
    creating build/temp.linux-aarch64-2.7
    creating build/temp.linux-aarch64-2.7/c
    aarch64-linux-gnu-gcc -pthread -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-Mlh1FU/python2.7-2.7.18=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python2.7 -c c/_cffi_backend.c -o build/temp.linux-aarch64-2.7/c/_cffi_backend.o
    c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
       15 | #include <ffi.h>
          |          ^~~~~~~
    compilation terminated.
    error: command 'aarch64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
    ERROR: Failed building wheel for cffi
 ```

## Solution

The Ubuntu image was missing the dependency `libffi-dev`. Installing this via aptitude and reinstalling pyload fixes this issue.

I'm not sure if we'd want to update the dependencies in the script, but I created this issue in case others encounter this in the future.

--- a/scripts/install/pyload.sh
+++ b/scripts/install/pyload.sh
@@ -28,6 +28,10 @@ else
     LIST='tesseract-ocr gocr rhino libcurl4-openssl-dev python2.7-dev sqlite3'
 fi
 
+if [[ $(_os_arch) =~ "arm" ]]; then
+    LIST+=' libffi-dev'
+fi
+
 apt_install $LIST
 
 if [[ ! $codename =~ ("xenial"|"stretch"|"buster"|"bionic") ]]; then

--- a/sources/functions/pyenv
+++ b/sources/functions/pyenv
@@ -80,7 +80,7 @@ function python_getpip() {
         echo_log_only "pip is available"
     fi
 
-    if ! ${python2bin} -m virtualenv > /dev/null 2>&1; then
+    if ! ${python2bin} -m virtualenv -h > /dev/null 2>&1; then
         echo_progress_start "Installing virtualenv"
         ${python2bin} -m pip install virtualenv >> "${log}" 2>&1
         echo_progress_done

--- a/sources/functions/pyenv
+++ b/sources/functions/pyenv
@@ -59,18 +59,31 @@ function pyenv_create_venv() {
 
 # Install pip appropriately if not yet installed
 function python_getpip() {
-    if ! command -v pip > /dev/null 2>&1; then
+    if command -v python2.7 > /dev/null 2>&1; then
+        python2bin=python2.7
+    elif command -v python2 > /dev/null 2>&1; then
+        python2bin=python2
+    else
+        echo_error "No python 2 installed"
+    fi
+
+    if ! ${python2bin} -m pip > /dev/null 2>&1; then
         echo_progress_start "Installing pip"
-        if command -v python2.7 > /dev/null 2>&1; then
-            python2bin=python2.7
-        else
-            python2bin=python2
-        fi
-        cd /tmp
+        cd /tmp || {
+            echo_error "Cannot cd to /tmp"
+            exit 1
+        }
         curl https://bootstrap.pypa.io/pip/2.7/get-pip.py -o get-pip.py >> $log 2>&1
         ${python2bin} get-pip.py >> $log 2>&1
-        pip install virtualenv >> "${log}" 2>&1
         echo_progress_done "Pip installed"
+    else
+        echo_log_only "pip is available"
+    fi
+
+    if ! ${python2bin} -m virtualenv > /dev/null 2>&1; then
+        echo_progress_start "Installing virtualenv"
+        ${python2bin} -m pip install virtualenv >> "${log}" 2>&1
+        echo_progress_done
     fi
 }
 

